UI core: text area

achrafmohye opened the pull request:
- Implements a text area component for capturing longer-form input.
- styling and alignment
- fix focus by Tab
- apply specific style on status-icon of filed wrapper
- fix focus by tab on input

achrafmohye committed:
ui-core: add textarea component

achrafmohye committed:
ui-core: fix pretttier

achrafmohye committed:
ui-core: re-arrange story

kmer2016 commented on ui-core/src/components/inputs/InputStatusIcon.tsx:
```diff
@@ -0,0 +1,30 @@
+import React from "react";
```
fix prettier error

kmer2016 commented on ui-core/src/components/inputs/TextArea.tsx:
```diff
@@ -0,0 +1,84 @@
+  const [value, setValue] = useState(initialValue);
+  const [focusViaKeyboard, setFocusViaKeyboard] = useState(false);
+
+  useKeyPress('Tab', async () => setFocusViaKeyboard(true));
```
This strategy to handle focus using Tab is buggy, there are an [issue](https://github.com/OpenRailAssociation/osrd-ui/issues/34) where it is described. And the fix is [here](https://github.com/OpenRailAssociation/osrd-ui/pull/35) . So you can reuse the same fix.

kmer2016 commented on ui-core/src/components/inputs/TextArea.tsx:
```diff
@@ -0,0 +1,84 @@
+  inputWrapperClassname?: string;
+};
+
+const Input: React.FC<InputProps> = ({
```
You forgot to rename the component into "TextArea".

kmer2016 commented on ui-core/src/components/inputs/TextArea.tsx:
```diff
@@ -0,0 +1,84 @@
+  const statusClassname = statusWithMessage?.status ? { [statusWithMessage.status]: true } : {};
+
+  return (
+    <div className={cx('text-area-wrapper', statusClassname)}>
```
There is a new component called FieldWrapper that can be used here. It includes the label, its style, and elements related to validation in a form. Please refer to the Input component to see how it is used.

kmer2016 commented on ui-core/src/styles/inputs/textArea.css:
```diff
@@ -0,0 +1,211 @@
+  to {
+    transform: rotate(360deg);
+  }
+}
```
It needs to be defined in a single place (either in the tailwind config or in a single CSS file) before being used everywhere it's needed.

achrafmohye committed:
Fix review comments

kmer2016 committed:
ui-core, text-area: - allow to override html attribute, use maxLength
- cleaning stories
- fix focus by Tab
- apply specific style on status-icon of filed wrapper
- fix focus by tab on input
